src/main/java/me/bymartrixx/vtd/gui/widget/PackListWidget.java

```diff
@@ -220,12 +229,33 @@ public class PackEntry extends EntryListWidget.Entry<PackListWidget.PackEntry> {
         public final String description;
         public final String[] incompatiblePacks;
 
+        private final Identifier icon;
+
         PackEntry(JsonObject pack) {
             this.name = pack.get("name").getAsString();
 
             this.displayName = pack.get("display").getAsString();
             this.description = StringUtils.normalizeSpace(pack.get("description").getAsString().replaceAll("<[^>]*>", " ")); // strip html tags from descriptions
 
+            this.icon = new Identifier(VTDMod.MOD_ID, this.name.toLowerCase());
+
+            if (MinecraftClient.getInstance().getTextureManager().getOrDefault(icon, null) == null) {
+                Thread iconDownloadThread = new Thread(() -> {
```

Comment:
Opening a thread for each icon doesn't sound very efficient. It'd probably be better to have one or two threads downlading every icon

Reply:
Singlethreaded:
![singlethreaded](https://user-images.githubusercontent.com/35371030/175095765-aef41beb-c86d-48b3-ada6-fed87791d18a.gif)

Multithreaded:
![multithreaded](https://user-images.githubusercontent.com/35371030/175095761-aef7a6e9-515b-4c41-97f5-31fea7c185d0.gif)

Reply:
It still doesn't convince me to open so many threads. Besides, not every icon is visible as soon as you open a tab, so downloading could be done once the user can see the entry. Instead of opening a lot of threads and never using them again, you could use a thread pool?

Reply:
Refactored to use a thread pool, let me know if you would prefer a different type of thread pool